modin/pandas/indexing.py

```diff
@@ -633,6 +633,25 @@ def __getitem__(self, key):
         if self.df.empty:
             return self.df._default_to_pandas(lambda df: df.loc[key])
 
+        # Limited support for slicing operations on loc, and predominantly
+        # only with positional mapping by translating it to an iloc. To
+        # support proper labels (sorted) we will need to translate the
+        # existing index to a mask OR translate the expression to a set of
+        # predicates.
+        if is_slice(key):
+            if key.step is not None:
+                raise NotImplementedError("Slice with step not supported for loc")
+            if key.start is None and key.stop is None:
+                return self.df 
+            if key.start is None:
+                new_index = (self.df.index <= key.stop)
+                return self.df.iloc[ new_index.start : ( new_index.stop + 1) ]
+            if key.stop is None:
+                new_index = (self.df.index >= key.start)
+                return self.df.iloc[ new_index.start : ( new_index.stop + 1) ]
+            # Works for a positional mapping
+            return self.df.iloc[ key.start : ( key.stop + 1) ]
```

Comment:
Is this necessary? IIRC, we should be handling slices without needing this change, right? Or is that support more general than we can support right now?

Reply:
No, we currently cannot handle slices at all because we don't have the index support for it. Some of this will change a bit as I implement support for this on non-default indexes.

Reply:
Ahhh I see, so that's why we're just forcing it to do the `iloc` here instead.